src/components/Modal.js

```diff
@@ -0,0 +1,126 @@
+            <label for="distance text-caption">거리(도보 이동 시간) </label>
+            <select name="distance" id="distance" required>
+              <option value="">선택해 주세요</option>
+              <option value="5">5분 내</option>
+              <option value="10">10분 내</option>
+              <option value="15">15분 내</option>
+              <option value="20">20분 내</option>
+              <option value="30">30분 내</option>
+            </select>
+          </div>
+
+          <div class="form-item">
+            <label for="description text-caption">설명</label>
+            <textarea name="description" id="description" cols="30" rows="5"></textarea>
+            <span class="help-text text-caption">메뉴 등 추가 정보를 입력해 주세요.</span>
+          </div>
+
+          <div class="form-item">
+            <label for="link text-caption">참고 링크</label>
+            <input type="text" name="link" id="link">
+            <span class="help-text text-caption">매장 정보를 확인할 수 있는 링크를 입력해 주세요.</span>
+          </div>
+
+          <div class="button-container">
+            <button type="button" class="button button--secondary text-caption">취소하기</button>
+            <button class="button button--primary text-caption">추가하기</button>
+          </div>
+        </form>
+      </div>
+    </div>
+    `;
+  }
+
+  render() {
+    if (!document.querySelector(`.modal`)) {
+      this.$target.insertAdjacentHTML('beforeend', this.template());
+    }
+
+    this.setSubmitEvent();
+    this.setModalCloseEvent();
+  }
+
+  setSubmitEvent() {
+    const $modalForm = document.querySelector('.modal form');
+
+    $modalForm.addEventListener('submit', e => {
+      e.preventDefault();
+
+      const category = document.querySelector('#category').value;
+      const name = document.querySelector('#name').value;
+      const distance = document.querySelector('#distance').value;
+      const description = document.querySelector('#description').value;
+      const link = document.querySelector('#link');
+
+      const restaurant = this.makeRestaurant({ category, name, distance, description, link });
+
+      this.changeRestaurantsState(restaurant);
+
+      document.querySelector('.modal').remove();
+    });
+  }
+
+  changeRestaurantsState(restaurant) {
+    this.restaurantsList.restaurants.add(restaurant);
+    localStorage.setItem('restaurants', JSON.stringify(this.restaurantsList.restaurants.restaurants));
+
+    this.restaurantsList.setState(restaurant);
+  }
+
+  setModalCloseEvent() {
+    const $cancelButton = document.querySelector('.modal .button--secondary');
+
+    $cancelButton.addEventListener('click', e => {
+      e.preventDefault();
+
+      document.querySelector('.modal').remove();
+    });
+  }
+
+  makeRestaurant(restaurant) {
+    const { category, name, distance, description, link } = restaurant;
+
+    return {
+      category: category,
+      name: name,
+      distance: distance,
+      description: description,
+      link: link,
+    };
```

Comment:
```suggestion
    return { category, name, distance, description, link };
```
필드명과 이름이 같다면 이렇게 내보내줘도 좋을거같아요

Reply:
이게 같은 것인지 몰랐네요.. 바꾸니 코드가 훨씬 간결해졌습니다!